src/library_webgl.js

```diff
@@ -1479,7 +1479,7 @@ var LibraryGL = {
       if (GLctx.currentPixelUnpackBufferBinding) {
         GLctx['compressedTexSubImage2D'](target, level, xoffset, yoffset, width, height, format, imageSize, data);
       } else {
-        GLctx['compressedTexSubImage2D'](target, level, xoffset, yoffset, width, height, format, HEAPU8, data, imageSize);
+        if (imageSize) GLctx['compressedTexSubImage2D'](target, level, xoffset, yoffset, width, height, format, HEAPU8, data, imageSize);
```

Comment:
Here I can maybe understand this if check - however I wish browsers could handle this problem of uploading a zero sized subimage, and this would be specced properly? Now we place a if check here because the browser behavior is not well defined?

Could this instead add a WebGL 2 conformance test to a defined behavior? (What I'd define the behavior to is to validate + then do a no-op if target/level/xoffset/yoffset/width/height/format all are good)